jscomp/ml/typecore.ml

```diff
@@ -1348,12 +1384,20 @@ and type_pat_aux ~constrs ~labels ~no_existentials ~mode ~explode ~env
         | _            -> k None
       end
   | Ppat_record(lid_sp_list, closed) ->
-      let opath, record_ty =
+      let has_dict_pattern_attr = Dicts.has_dict_pattern_attribute sp.ppat_attributes in
+      let opath, record_ty = (
+      match (has_dict_pattern_attr, expected_ty.desc) with 
+      | (true, Tvar _) -> 
+        (* When this is a dict pattern and we don't have an actual expected type yet, 
```

Comment:
how do we know it does not have an expected type yet?

Reply:
why not make zero assumptions and simply rely on unification with always creating a fresh var?